src/interfaces/IAppliance.js

```diff
@@ -0,0 +1,95 @@
+import {
+	InterfaceInstantiationError,
+	NotImplementedError,
+} from '../errors'
+
+class IAppliance {
+	/**
+	 * Constructor for a new appliance
+	 * @param  {Object} overrideSettings Values to override default appliance settings.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	constructor(overrideSettings = {}) {
+		if (this.constructor === IAppliance) {
+			throw new InterfaceInstantiationError(this.constructor.name)
+		}
+	}
+
+	/**
+	 * Getter for the list of data types accepted by the appliance.
+	 *
+	 * @return {String[]} The list of data types accepted by the appliance.
+	 */
+	getInputTypes = () => {
+		throw new NotImplementedError('getInputTypes')
+	}
+
+	/**
+	 * Getter for the list of data types produced by the appliance.
+	 *
+	 * @return {String[]} The list of data types produced by the appliance.
+	 */
+	getOutputTypes = () => {
+		throw new NotImplementedError('getOutputTypes')
+	}
+
+	/**
+	 * Asserts that a given payload is actually a payload.
+   *
+	 * @param  {Payload} payload The payload that we want to validate.
+	 * @return {Boolean}         The result of the validation check.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	isValidPayload = async (payload) => {
+		throw new NotImplementedError('isValidPayload')
+	}
+
+	/**
+	 * Invokes the appliance on any unprocessed data in the appliance buffer.
+	 *
+	 * @return {Boolean} Whether or not the appliance had enough data to work with.
+	 */
+	invoke = async () => {
+		throw new NotImplementedError('invoke')
+	}
+
+	/**
+	 * Called to pass data into the appliance.
+	 * If the payload is valid, it is added to the buffer and the appliance is invoked.
+	 *
+	 * @throws AssertionError when passed an invalid payload for this appliance.
```

Comment:
Quibble: Feel like we should put `@throws` at the end, after `@return`? Like while reading this function definition I'd want to know "what does it accept, what does it return, and what errors might it throw if something goes wrong". I'd not want to start with the sadness.